swarm/network/simulation/kademlia.go

```diff
@@ -94,3 +96,53 @@ func (s *Simulation) kademlias() (ks map[enode.ID]*network.Kademlia) {
 	}
 	return ks
 }
+
+func (s *Simulation) WaitNetworkHealth() error {
+	// construct the peer pot, so that kademlia health can be checked
+	addrs := make([][]byte, len(s.Net.Nodes))
+	ids := make([]enode.ID, len(s.Net.Nodes))
+	for i, v := range s.Net.Nodes {
+		ids[i] = v.ID()
+		addrs[i] = v.ID().Bytes()
+	}
+	// construct the peer pot, so that kademlia health can be checked
+	ppmap := network.NewPeerPotMap(2, addrs) //yes i'm hardcoding this
+	check := func(ctx context.Context, id enode.ID) (bool, error) {
+		select {
+		case <-ctx.Done():
+			return false, ctx.Err()
+		default:
+		}
+
+		node := s.Net.GetNode(id)
+		if node == nil {
+			return false, fmt.Errorf("unknown node: %s", id)
+		}
+		client, err := node.Client()
+		if err != nil {
+			return false, fmt.Errorf("error getting node client: %s", err)
+		}
+		healthy := &network.Health{}
+		if err := client.Call(&healthy, "hive_healthy", ppmap[id.String()]); err != nil {
```

Comment:
I believe we should have a proxy method in `hive` that returns analyzed results based on certain inputs. Evaluating this components of the `Health` object on the client side does not seem correct to me.

Should that be part of this PR?

Reply:
what would be the inputs?